pkgs/applications/video/vivictpp/default.nix

```diff
@@ -0,0 +1,71 @@
+{ lib, stdenv, fetchFromGitHub
+, meson, cmake, ninja, pkg-config
+, python3, git
+, SDL2, SDL2_ttf
+, freetype
+, ffmpeg
+, cacert }:
+
+let
+  withSubprojects = stdenv.mkDerivation {
+    name = "sources-with-subprojects";
+
+    src = fetchFromGitHub {
+      owner = "vivictorg";
+      repo = "vivictpp";
+      rev = "f682bec15cf69faa3c37b8fc5e27408b89da2281";
+      hash = "sha256-hb5LCPsoda27oCJrhZaWq3ov+SmMLdECXvTXfrXFJgo=";
+    };
+
+    nativeBuildInputs = [
+      meson
+      cacert
+      git
+    ];
+
+    buildCommand = ''
+      cp -r --no-preserve=mode $src $out
+      cd $out
+
+      meson subprojects download
+      find subprojects -type d -name .git -prune -execdir rm -r {} +
+    '';
+
+    outputHashMode = "recursive";
+    outputHash = "sha256-J7B9l03r+D09a1mj6aChZfLHgYfNOJUHIpPr5LORKAE=";
```

Comment:
This is the sketchiest bit of the PR. It... *should* be stable, but it is still fragile to bunch up multiple sources like this. The alternative (interpreting [`.wrap` files](https://github.com/vivictorg/vivictpp/tree/f682bec15cf69faa3c37b8fc5e27408b89da2281/subprojects) and placing the subprojects with nix) would need more reusable nixpkgs support though.